pandas/core/indexes/datetimelike.py

```diff
@@ -164,22 +164,6 @@ def __contains__(self, key: Any) -> bool:
             is_scalar(res) or isinstance(res, slice) or (is_list_like(res) and len(res))
         )
 
-    # Try to run function on index first, and then on elements of index
-    # Especially important for group-by functionality
-    def map(self, mapper, na_action=None):
```

Comment:
@jbrockmendel : do you have any issues with moving the `map` definition in `DatetimeIndexOpsMixin` down to `ExtensionIndex` so it's shared with `IntervalIndex`?  This seems to more gracefully handle index specific attributes (e.g. `freq` for DTI, `closed` for II) than `Index.map`.

Reply:
definitely on board with this if we can use this for the general case